### PR TITLE
docs: adding export metadata docs for light client module genesis

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Light client guide `ExportMetadata`

closes: #1857 

### Commit Message / Changelog Entry

```bash
docs: adding export metadata docs for light client module genesis
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/ibc/light-clients/genesis.md [hunk 1]
```diff
@@ -1,3 +1,35 @@
...
+
+`ClientState` instances are provided their own isolated and namespaced client store upon initialisation. `ClientState` implementations may choose to store any amount of arbitrary metadata in order to perform light client operations correctly. 
+
+The `ExportMetadata` method of the `ClientState` interface provides light client modules with the ability to persist metadata in genesis exports. 
+
+```go
+ExportMetadata(clientStore sdk.KVStore) []GenesisMetadata
+```
+
+`ExportMetadata` is provided the client store and returns an array of `GenesisMetadata`. For maximum flexibility, `GenesisMetadata` is defined as a simple interface containing two distinct `Key` and `Value` accesor methods.
```
Comment (charleenfei) [inline docs/ibc/light-clients/genesis.md:21]:
```suggestion
`ExportMetadata` is provided the client store and returns an array of `GenesisMetadata`. For maximum flexibility, `GenesisMetadata` is defined as a simple interface containing two distinct `Key` and `Value` accessor methods.
```

File: docs/ibc/light-clients/genesis.md [hunk 2]
```diff
@@ -1,3 +1,35 @@
...
 order: 10
--->
\ No newline at end of file
+-->
+
+# Genesis Metadata
+
+Learn how to implement the `ExportMetadata` interface {synopsis}
+
+## Pre-requisites Readings
```
Comment (charleenfei) [inline docs/ibc/light-clients/genesis.md:9]:
```suggestion
## Pre-requisite Readings
```

File: docs/ibc/light-clients/genesis.md [hunk 3]
```diff
@@ -1,3 +1,35 @@
...
+
+- [Cosmos SDK module genesis](https://docs.cosmos.network/v0.47/building-modules/genesis) {prereq}
+
+`ClientState` instances are provided their own isolated and namespaced client store upon initialisation. `ClientState` implementations may choose to store any amount of arbitrary metadata in order to perform light client operations correctly.
```
Comment (charleenfei) [inline docs/ibc/light-clients/genesis.md:13]: mega nit, but maybe a more precise wording for "perform light client operations" is "verify the counterparty consensus state"?

**Review by charleenfei — APPROVED** [review]
just a few minor nits but otherwise lgtm!

File: docs/ibc/light-clients/genesis.md (hunk 3 above)
Comment (damiannolan) [reply docs/ibc/light-clients/genesis.md]: I changed to:

> in order to verify counterparty consensus state and perform light client updates correctly

**Review by chatton — APPROVED** [review]
LGTM 👍

**Pull request merged**